README.md

```diff
@@ -0,0 +1,41 @@
+# fds-coreui-components-react
+React wrappers for Fintraffic Design System's [Core UI Components](https://github.com/fintraffic-design/fds-coreui-components).
+
+# Install
+
+```shell
+npm i --save @fintraffic/fds-coreui-components-react
+```
+
+# Usage
+
+Usage of React component wrappers.
+
+The simplest way to import and register a React component is importing it directly from the `@fintraffic/fds-coreui-components-react` package.
+
+```js
+import { FdsButton, FdsButtonVariant } from '@fintraffic/fds-coreui-components-react'
+```
+
+```jsx
+<FdsButton
+  label="My Button"
+  icon="plus"
+  variant={FdsButtonVariant.primary}
+  onClick={handleMyButtonClick}
+/>
+```
+
+However this will cause all of the components to be loaded and registered (not just the ones you import).
```

Comment:
Wouldn't the unused components get tree shaked during build or am I missing something?

Reply:
With the current setup at least where the custom element definitions are done implicitly as a side effect of the component import, using any of the components from the bare module import (that goes through `index.ts` that has a bunch of `export * from './<omponent>.js'`) it will trigger custom element registration of all the components.

If you import something from the package without using _anything_, then they won't be registered, but as soon as you use _something_ from there it will register all the custom elements. I have tested that this is what actually happens. You can check by seeing if you get a function instead of undefined when running for example `customElements.get('fds-dropdown')` after you only imported and used `FdsButton`.

It would be good if we can get nice short imports with better tree shaking but we can try to improve on that later. That could be done by making custom element registrations explicit (not automatic) but then the user would always need to call registration separately (once for all the used components).

We could make component import paths slightly nicer by using [subpath exports](https://nodejs.org/api/packages.html#subpath-exports) in `package.json` but that would still mean importing each component from different path.

For now these React wrappers are at least easy to use and auto-importing from at least VS Code seem to work nicely.

Reply:
Huh, that is news to me, but yes this atleast a good start. 👍 I will say though, that most devs will probably end up importing like this:
`import { FdsButton, FdsButtonVariant } from '@fintraffic/fds-coreui-components-react'`
without understanding that this basically will then bundle every component, since realistically not everyone will read the README. It's probably not going to have any noticeably effect however as the library is still quite small, so it's really a non-issue, just something to be aware of.